notion-core/src/main/scala/com/ambiata/notion/core/StoreTemporary.scala

```diff
@@ -0,0 +1,38 @@
+package com.ambiata.notion.core
+
+import com.ambiata.mundane.control._
+import com.ambiata.mundane.io._
+import com.ambiata.notion.core.{TemporaryType => T}
+import com.ambiata.poacher.hdfs._
+import com.ambiata.poacher.hdfs.TemporaryConfiguration._
+import com.ambiata.saws.core.Clients
+import com.ambiata.saws.s3._
+import org.apache.hadoop.conf.Configuration
+
+import scalaz._, Scalaz._
+
+import scalaz.{Store => _}
+
+case class StoreTemporary(t: T, path: String) {
+  val client = Clients.s3
+
+  def store: RIO[Store[RIO]] =
+    if (t == T.Posix)     posixStore
+    else if (t == T.S3)   s3Store
+    else if (t == T.Hdfs) hdfsStore
+    else                  RIO.fail("Could not create store. Unsupported temporary type")
```

Comment:
Can't you pattern match on `t` this isn't possible and the compiler can check that right?